latest `nightly` release fails in CI during foundryup

### Component

Foundryup

### Have you ensured that all of these are up to date?

- [ ] Foundry
- [x] Foundryup

### What version of Foundry are you on?

_No response_

### What version of Foundryup are you on?

_No response_

### What command(s) is the bug in?

_No response_

### Operating System

Linux

### Describe the bug

Our CI recently started failing with this error while installing foundry nightly via the `foundry-rs/foundry-toolchain@v1` github action:

```
. prepare: forge: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.39' not found (required by forge)
. prepare: forge: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by forge)
```

(See https://github.com/latticexyz/mud/actions/runs/13909195313/job/38919307523#step:5:119)

It works correctly when pinning foundry to version `nightly-f47d7e0c29a36372908b917cd74aa589d5888f8e`. (See https://github.com/latticexyz/mud/pull/3628)

Why do you think it's `pnpm install --frozen-lockfile` that's failing? This the output i see right before the error:

```
prepare$ husky install && (forge --version || pnpm foundryup)
. prepare: husky - Git hooks installed
. prepare: forge: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.39' not found (required by forge)
. prepare: forge: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by forge)
. prepare: forge: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.38' not found (required by forge)
. prepare: > mud-monorepo@ foundryup /home/runner/work/mud/mud
. prepare: > curl -L https://foundry.paradigm.xyz/ | bash && bash ~/.foundry/bin/foundryup
. prepare:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
. prepare:                                  Dload  Upload   Total   Spent    Left  Speed
. prepare:   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
. prepare: 100   167  100   167    0     0   2734      0 --:--:-- --:--:-- --:--:--  2737
. prepare: 100  2196  100  2196    0     0  17997      0 --:--:-- --:--:-- --:--:-- 17997
. prepare: Installing foundryup...
. prepare: Detected your preferred shell is bash and added foundryup to PATH.
. prepare: Run 'source /home/runner/.bashrc' or start a new terminal session to use foundryup.
. prepare: Then, simply run 'foundryup' to install Foundry.
. prepare: bash: /home/runner/.foundry/bin/foundryup: No such file or directory
. prepare:  ELIFECYCLE  Command failed.
. prepare: Failed
 ELIFECYCLE  Command failed with exit code 1.
Error: Process completed with exit code 1.
```

This is running on a depot.dev with `ubuntu-22.04-16`.

The reason why I think it has to do with a recent nightly release is that it works without issues when pinning to `nightly-f47d7e0c29a36372908b917cd74aa589d5888f8e` from 2 days ago (using the same runner and same repo on our side, see https://github.com/latticexyz/mud/pull/3628)

--- a/.github/workflows/docker-publish.yml
+++ b/.github/workflows/docker-publish.yml
@@ -29,10 +29,10 @@ jobs:
       - uses: Swatinem/rust-cache@v2
         with:
           cache-on-failure: true
-      - name: Install cross
-        id: cross
+      - name: Install cross main
+        id: cross_main
         run: |
-          cargo install cross
+          cargo install cross --git https://github.com/cross-rs/cross
       # Login against a Docker registry except on PR
       # https://github.com/docker/login-action
       - name: Login into registry ${{ env.REGISTRY }}

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -89,22 +89,22 @@ jobs:
           # `target`: Rust build target triple
           # `platform` and `arch`: Used in tarball names
           # `svm`: target platform to use for the Solc binary: https://github.com/roynalnaruto/svm-rs/blob/84cbe0ac705becabdc13168bae28a45ad2299749/svm-builds/build.rs#L4-L24
-          - runner: Linux-22.04
+          - runner: Linux-20.04
             target: x86_64-unknown-linux-gnu
             svm_target_platform: linux-amd64
             platform: linux
             arch: amd64
-          - runner: Linux-22.04
+          - runner: Linux-20.04
             target: x86_64-unknown-linux-musl
             svm_target_platform: linux-amd64
             platform: alpine
             arch: amd64
-          - runner: Linux-22.04
+          - runner: Linux-20.04
             target: aarch64-unknown-linux-gnu
             svm_target_platform: linux-aarch64
             platform: linux
             arch: arm64
-          - runner: Linux-22.04
+          - runner: Linux-20.04
             target: aarch64-unknown-linux-musl
             svm_target_platform: linux-aarch64
             platform: alpine

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -152,7 +152,7 @@ jobs:
       - name: MUSL setup
         if: matrix.target == 'x86_64-unknown-linux-musl' || matrix.target == 'aarch64-unknown-linux-musl'
         run: |
-          cargo install cross --git https://github.com/cross-rs/cross
+          cargo install cross
 
       - name: Build binaries
         env:

--- a/crates/forge/bin/cmd/bind_json.rs
+++ b/crates/forge/bin/cmd/bind_json.rs
@@ -104,7 +104,7 @@ impl BindJsonArgs {
                 let ast = parser.parse_file().map_err(|e| e.emit())?;
 
                 let mut visitor = PreprocessorVisitor::new();
-                visitor.visit_source_unit(&ast);
+                let _ = visitor.visit_source_unit(&ast);
                 visitor.update(&sess, &mut content);
 
                 source.content = Arc::new(content);

--- a/crates/forge/bin/cmd/geiger.rs
+++ b/crates/forge/bin/cmd/geiger.rs
@@ -129,7 +129,7 @@ fn try_lint_file(
     let mut parser = solar_parse::Parser::from_file(sess, &arena, path)?;
     let ast = parser.parse_file().map_err(|e| e.emit())?;
     let mut visitor = Visitor::new(sess, unsafe_cheatcodes);
-    visitor.visit_source_unit(&ast);
+    let _ = visitor.visit_source_unit(&ast);
     Ok(visitor.count)
 }
 

